infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -329,34 +348,84 @@ private void connectToClusterManager() {
-        // such as streams to replicate and version
-        LogReplicationConfig logReplicationConfig = getLogReplicationConfiguration(getCorfuRuntime());
 
-        logReplicationMetadataManager = new LogReplicationMetadataManager(getCorfuRuntime(),
-            topologyDescriptor.getTopologyConfigId(), localClusterDescriptor.getClusterId());
+        if (role != ClusterRole.ACTIVE && role != ClusterRole.STANDBY) {
+            log.debug("Cluster role is {}.  Not bootstrapping further.", role);
+            return;
+        }
 
-        logReplicationServerHandler = new LogReplicationServer(serverContext, logReplicationConfig,
-            logReplicationMetadataManager, localCorfuEndpoint, topologyDescriptor.getTopologyConfigId(), localNodeId);
-        logReplicationServerHandler.setActive(localClusterDescriptor.getRole().equals(ClusterRole.ACTIVE));
-        logReplicationServerHandler.setStandby(localClusterDescriptor.getRole().equals(ClusterRole.STANDBY));
+        performRoleBasedSetup(role);
+        bootstrapComplete = true;
+    }
 
-        interClusterReplicationService = new CorfuInterClusterReplicationServerNode(serverContext,
-            logReplicationServerHandler, logReplicationConfig);
+    private void performRoleBasedSetup(ClusterRole role) {
 
-        // Pass server's channel context through the Log Replication Context, for shared objects between the server
-        // and the client channel (specific requirements of the transport implementation)
-        replicationContext = new LogReplicationContext(logReplicationConfig, topologyDescriptor,
-            localCorfuEndpoint, interClusterReplicationService.getRouter().getServerAdapter().getChannelContext());
+        // Through LogReplicationConfigAdapter retrieve system-specific configurations
+        // such as streams to replicate and version
+        LogReplicationConfig logReplicationConfig =
+            getLogReplicationConfiguration(getCorfuRuntime());
+
+        Set<String> remoteClusterIds = new HashSet<>();
+
+        if (role == ClusterRole.ACTIVE) {
+            remoteClusterIds.addAll(topologyDescriptor.getStandbyClusters()
+                .keySet());
+            createMetadataManagers(remoteClusterIds);
+            replicationContext = new LogReplicationContext(logReplicationConfig,
+                topologyDescriptor, localCorfuEndpoint);
+            log.info("Replication context {}", replicationContext);
+            logReplicationEventListener = new LogReplicationEventListener(this,
+                getCorfuRuntime());
+            logReplicationEventListener.start();
+        }
 
-        // Unblock server initialization & register to Log Replication Lock, to attempt lock / leadership acquisition
-        serverCallback.complete(interClusterReplicationService);
+        if (role == ClusterRole.STANDBY) {
+            remoteClusterIds.addAll(topologyDescriptor.getActiveClusters()
```

Comment:
yes, we can have the behavior where the Source connects to a subset of the Sink clusters.  For the new replication models, it will be tied to the replication config we receive - for example: Sink-x does not belong to any replication group so no need to establish connection with it.  Without that info, we cannot determine whom to connect to.